packages/sitecore-jss/src/personalize/layout-personalizer.ts

```diff
@@ -86,7 +101,14 @@ export function personalizeComponent(
     return null;
```

Comment:
Need to apply the same logic here as well (hidden by default case) and add accompanying unit test. 
I just tested this and it suffers from the same issue.